tests/sentry/snuba/metrics/test_mqb_query_transformer.py

```diff
@@ -0,0 +1,707 @@
+                    metric_name="d:transactions/measurements.fid@millisecond",
+                    args=["meh"],
+                    alias="count_web_vitals_measurements_fid_meh",
+                ),
+                MetricField(
+                    op="count_web_vitals_measurements",
+                    metric_name="d:transactions/measurements.fp@millisecond",
+                    args=["good"],
+                    alias="count_web_vitals_measurements_fp_good",
+                ),
+                MetricField(
+                    op="count_web_vitals_measurements",
+                    metric_name="d:transactions/measurements.lcp@millisecond",
+                    args=["good"],
+                    alias="count_web_vitals_measurements_lcp_good",
+                ),
+                MetricField(
+                    op="count_web_vitals_measurements",
+                    metric_name="d:transactions/measurements.fcp@millisecond",
+                    args=["meh"],
+                    alias="count_web_vitals_measurements_fcp_meh",
+                ),
+            ],
+            start=datetime.datetime(2022, 3, 24, 11, 11, 35, 447729, tzinfo=pytz.utc),
+            end=datetime.datetime(2022, 6, 22, 11, 11, 35, 447729, tzinfo=pytz.utc),
+            granularity=Granularity(granularity=3600),
+            where=None,
+            groupby=[MetricGroupByField("transaction", alias=None)],
+            limit=Limit(limit=51),
+            offset=Offset(offset=0),
+            include_series=False,
+        ),
+        id="count_web_vitals test case",
+    ),
+    pytest.param(
+        Query(
+            match=Entity("generic_metrics_distributions"),
+            select=[
+                Function(
+                    function="p95",
+                    initializers=None,
+                    parameters=[Column("d:transactions/duration@millisecond")],
+                    alias="p95",
+                ),
+                Function(
+                    function="rate",
+                    initializers=None,
+                    parameters=[
+                        Column("d:transactions/duration@millisecond"),
+                        60,
+                        7776000.0,
+                    ],
+                    alias="epm",
+                ),
+                Function(
+                    function="e:transaction/failure_rate@ratio",
+                    initializers=None,
+                    parameters=[],
+                    alias="failure_rate",
+                ),
+            ],
+            groupby=[
+                AliasedExpression(
+                    exp=Column("transaction.status"),
+                    alias="transaction.status",
+                ),
+                Function(
+                    function="e:transactions/team_key_transactions@none",
+                    initializers=None,
+                    parameters=[
+                        [(13, "foo_transaction")],
```

Comment:
I'm having a hard time understanding what this means. Is the intention to group by "is key transaction of project 13"? If so, what does the `"foo_transaction"` argument mean?

Reply:
so key transaction is basically `tuple(project_id, transaction) IN [(13, foo_transaction)]` so second argument is transaction name

Reply:
Ah ok, so `foo_transaction` _is_ a key transaction. Does it even make sense to create a derived metric for this then? I would have assumed that the derived metric queries the key transactions from postgres under the hood, but if the creator of the `Query` has to pass those explicitly, we might as well have a more generic

```python
Function("has", ["foo_transaction"], Column("tags[transaction]"), alias="is_key_transaction")
```

Reply:
Sorry I am not following your example fully, so is team key transaction  checks on both project and transaction name
We could potentially add that as a condition but then it would break our select/orderby because we don't allow conditions in select or orderby and that is why I thought it might be better for just this one case to create a derived metric as I don't believe we have a need for something like this otherwise

Reply:
+1, a key transaction is a tuple of project&transaction cause you could easily have a transaction named `transaction_foo` in `project_A`, and another `transaction_foo` in `project_B` but they're completely different transactions